pkg/controller/vitesskeyspace/reconcile_shards.go

```diff
@@ -152,6 +153,17 @@ func (r *reconcileHandler) reconcileShards(ctx context.Context) error {
 	for i := range r.vtk.Status.Partitionings {
 		status := &r.vtk.Status.Partitionings[i]
 		status.ServingWrites = allShardsServingWrites(status.ShardNames, r.vtk.Status.Shards)
+		status.Tablets = totalTablets(status.ShardNames, r.vtk.Status.Shards)
+		status.ReadyTablets = totalReadytablets(status.ShardNames, r.vtk.Status.Shards)
+		status.UpdatedTablets = totalUpdatedTablets(status.ShardNames, r.vtk.Status.Shards)
+
+		if status.ServingWrites == corev1.ConditionTrue {
```

Comment:
If we get through the loop without finding any serving partitioning, we should set the Ready condition status to False with reason NoServingPartitioning. Otherwise it'll be left at Unknown/ReconcileFailed which doesn't represent what we actually know in that case.

Reply:
Done.